### DISCUSSION TITLE
React Native

### DISCUSSION THREAD

**Discussion by 3ops-jgheeraert** [refinedev/refine#5540] [Integration request]
I would like to use refine to create an app that can also work offline. I was thinking of using refine with react native as the platform. Is this already supported, or is this in the pipeline? I found somewhere that it would be supported in version 4, but find very little documentation on this. Is this coming, will templates be made,...? Since I currently have little react experience, templates and examples would help me.

Thanks in advance for all the tips and info!

**Comment by BatuhanW** [comment]
Hello @3ops-jgheeraert, we have a community package for react-native: https://dev.to/abdellah711/refine-native-2p1d

IF you have issues with offline, you can check this comment: https://github.com/refinedev/refine/issues/5510#issuecomment-1897914457

**Reply by 3ops-jgheeraert** [reply]
Thanks! But I found this project already yesterday, but was not to able to make it running.

I think I followed the steps described on the project page well.

1. creating a new expo project, described here: https://docs.expo.dev/router/installation/#manual-installation
2. then added the following packages like request on the project page: yarn add @refinenative/expo-router @refinenative/react-native-paper @refinedev/simple-rest @refinedev/core
3. add the refine code as described on the project page.
4. run the project

But then I get following error: 
**_Unable to resolve "react-native-paper" from "node_modules\@refinenative\react-native-paper\dist\index.js"_**

I installed the react-native-paper package.

Any ideas or suggestions?

**Reply by BatuhanW** [reply]
Hello @3ops-jgheeraert, I think you need to `also` install `react-native-paper` package separately.

`yarn add react-native-paper`.

**Reply by 3ops-jgheeraert** [reply]
Thanks. Did this, but then I get error: Unable to resolve "@react-navigation/drawer" from "node_modules\expo-router\build\layouts\Drawer.js"

I did then yarn add react-navigation/drawer, but the error is the same afterwards.

**Reply by BatuhanW** [reply]
You can try with `yarn add @react-navigation/drawer`.

**Reply by 3ops-jgheeraert** [reply]
Yep, tried this already. Then an error on non existing react-native-reanimated. Install this one, now another strange error.

The project's last commit is now 7 months ago, so either the manual is wrong, or I'm doing something completely wrong, or too many changes have been made in the meantime for this to work.

I'm going to park it for now and look at refine in function of next.js platform (without offline version then).

Anyway, thanks for the effort!
